### ISSUE TITLE
[Bug] DescribeInstance command does not provide microVM state

### ISSUE THREAD

**Issue by acatangiu** [firecracker-microvm/firecracker#2285]
## Describe the bug

Current `describeInstance` API command does not provide information about microVm state other than whether it's been started or not. Starting with `v0.23.0` the microVm can be also `paused` and this state information is lost to the user.

## To Reproduce

```
sudo curl -s --unix-socket api.socket "http://localhost/"
{"id":"27","started":false,"vmm_version":"0.23.0","app_name":"Firecracker"}
```

## Expected behaviour

Turn the `started` boolean into a multiple options data struct with variants like:
 - Not started
 - Running
 - Paused

```
sudo curl -s --unix-socket api.socket "http://localhost/"
{"id":"27","state":"Running","vmm_version":"0.23.0","app_name":"Firecracker"}
```

**Comment by HQ01** [comment]
@acatangiu do we also need to change any configuration/documentation file after implementing this change?

**Comment by acatangiu** [comment]
From a high-level look, changes required:

### Code

 - in `src/vmm/src/vmm_config/instance_info.rs`: change `started: bool` to `state: String` (or a `state: enum MicroVmState` that implements ToString)
 - replace `started` with `state` everywhere
 - in `src/vmm/src/rpc_interface.rs` change `state` accordingly on `Pause` and `Resume` commands.

### Config/docs

 - `src/api_server/swagger/firecracker.yaml` change `InstanceInfo` definition to describe the new API
 - `CHANGELOG.md` describe the api change

**Comment by pkj415** [comment]
@acatangiu Can I take up this issue?

**Comment by HQ01** [comment]
I've started working on it.

**Comment by HQ01** [comment]
Implemented, will submit a PR after cleaning code up

**Comment by dianpopa** [comment]
Fixed by #2292.
